### PR TITLE
[Tests-Only] Refactor test using IntegrationTestTrait

### PR DESCRIPTION
**Pull request by swoichha**
### Description
Refactored `ProductsFrontendControllerTest.php` and `ManufacturersFrontendControllerTest.php` using assert method of `IntegrationTestTrait`.

### CODE REVIEW

File: tests/TestCase/src/Controller/ProductsFrontendControllerTest.php [hunk 1]
```diff
@@ -33,78 +35,77 @@ public function testProductDetailOfflineManufacturerPublicLoggedOut()
...
 
     public function testProductDetailOnlineManufacturerPublicLoggedOut()
     {
-        $this->httpClient->followOneRedirectForNextRequest();
-        $response = $this->httpClient->get($this->Slug->getProductDetail(60, 'Demo Product'));
-        $this->assertDoesNotMatchRegularExpressionWithUnquotedString('0,62 €', $response->getStringBody()); // price must not be shown
-        $this->assert200OkHeader();
+        $this->get($this->Slug->getProductDetail(60, 'Demo Product'));
+        $this->get($this->_response->getHeader('Location')[0]);
+        $this->assertDoesNotMatchRegularExpressionWithUnquotedString('0,62 €', $this->_response->getBody()); // price must not be shown
```
Comment (individual-it) [inline tests/TestCase/src/Controller/ProductsFrontendControllerTest.php:55]: what about https://api.cakephp.org/3.7/class-Cake.TestSuite.IntegrationTestTrait.html#__getBodyAsString

File: tests/TestCase/src/Controller/ProductsFrontendControllerTest.php [hunk 2]
```diff
@@ -33,78 +35,77 @@ public function testProductDetailOfflineManufacturerPublicLoggedOut()
...
     }
 
     public function testProductDetailOnlineManufacturerPublicLoggedIn()
     {
         $this->loginAsCustomerWithHttpClient();
-        $this->httpClient->followOneRedirectForNextRequest();
-        $this->httpClient->get($this->Slug->getProductDetail(60, 'Demo Product'));
-        $this->assert200OkHeader();
+        $this->get($this->Slug->getProductDetail(60, 'Demo Product'));
+        $this->get($this->_response);
```
Comment (individual-it) [inline tests/TestCase/src/Controller/ProductsFrontendControllerTest.php:72]: why is this needed?

File: tests/TestCase/src/Controller/ProductsFrontendControllerTest.php (hunk 1 above)
Comment (swoichha) [reply tests/TestCase/src/Controller/ProductsFrontendControllerTest.php:55]: Using `_getBodyAsString()` returned an error message `Error: Call to undefined method Cake\Http\Response::_getBodyAsString()` so i used `getBody()` instead.

File: tests/TestCase/src/Controller/ProductsFrontendControllerTest.php (hunk 2 above)
Comment (swoichha) [reply tests/TestCase/src/Controller/ProductsFrontendControllerTest.php:72]: Yes it is needed because the first `get` request doesn't work completely as it redirects to `/produkt/60-Demo-Product` while it should have been redirected to the `http://localhost/produkt/60-Milch`. So, using the second `get` request completes the work.

File: tests/TestCase/src/Controller/ProductsFrontendControllerTest.php (hunk 1 above)
Comment (individual-it) [reply tests/TestCase/src/Controller/ProductsFrontendControllerTest.php:55]: I believe it should be `$this->_getBodyAsString()`

File: tests/TestCase/src/Controller/ManufacturersFrontendControllerTest.php [hunk 3]
```diff
@@ -31,42 +33,41 @@ public function setUp(): void
 
        public function testManufacturerDetailOnlinePublicLoggedOut()
     {
-        $this->httpClient->followOneRedirectForNextRequest();
-        $this->httpClient->get($this->Slug->getManufacturerDetail(4, 'Demo Manufacturer'));
-        $this->assert200OkHeader();
+           $this->get($this->Slug->getManufacturerDetail(4, 'Demo Manufacturer'));
```
Comment (individual-it) [inline tests/TestCase/src/Controller/ManufacturersFrontendControllerTest.php:36]: wrong indent
